tests/round_trip.rs

```diff
@@ -0,0 +1,720 @@
+// Derived from object's round_trip.rs:
+// https://github.com/gimli-rs/object/blob/0.32.0/tests/round_trip/mod.rs
+
+use std::io::Cursor;
+
+use ar_archive_writer::{ArchiveKind, NewArchiveMember};
+use object::read::{Object, ObjectSection, ObjectSymbol};
+use object::{read, write};
+use object::{
+    Architecture, BinaryFormat, Endianness, RelocationEncoding, RelocationKind, SectionKind,
+    SymbolFlags, SymbolKind, SymbolScope, SymbolSection,
+};
+
+fn round_trip_via_archive_writer(
+    object: write::Object<'_>,
+    archive_kind: ArchiveKind,
+    check_func: impl FnOnce(read::File),
+) {
+    // Create a new archive using ar_archive_writer.
+    let input_member = NewArchiveMember {
+        buf: Box::new(object.write().unwrap()) as Box<dyn AsRef<[u8]>>,
+        get_symbols: ar_archive_writer::get_native_object_symbols,
+        member_name: "test.o".to_string(),
+        mtime: 0,
+        uid: 0,
+        gid: 0,
+        perms: 0o644,
+    };
+    let mut output_bytes = Cursor::new(Vec::new());
+    ar_archive_writer::write_archive_to_stream(
+        &mut output_bytes,
+        &[input_member],
+        true,
+        archive_kind,
+        true,
+        false,
+    )
+    .unwrap();
+
+    // Read the archive and member using object.
+    let output_bytes = output_bytes.into_inner();
+    let output_archive = read::archive::ArchiveFile::parse(output_bytes.as_slice()).unwrap();
+    let output_member = output_archive.members().next().unwrap().unwrap();
+    assert_eq!(output_member.name(), b"test.o");
+    let output_object =
+        read::File::parse(output_member.data(output_bytes.as_slice()).unwrap()).unwrap();
+
+    // Run the test-specific check.
+    check_func(output_object);
```

Comment:
Just checking that the output object is byte-for-byte  identical to the input object would be enough, right?

Reply:
Let me give it a try - even for the llvm-ar tests, I wasn't sure if we wanted to try byte-for-byte comparisons, or to use a tool like llvm-nm to give give some more context and avoid anything unstable.

Reply:
Done - although I'm seeing some weird behavior for 32-bit MachO...